src/Stripe.net/Infrastructure/Public/StripeConfiguration.cs

```diff
@@ -36,98 +85,95 @@ public static string StripeApiVersion
         /// otherwise the library will no longer be able to deserialize polymorphic resources
         /// represented by interfaces (e.g. <see cref="IPaymentSource"/>).
         /// </summary>
-        public static JsonSerializerSettings SerializerSettings
-        {
-            get { return serializerSettings; }
-            set { serializerSettings = value; }
-        }
-
-        public static string StripeNetVersion { get; }
+        public static JsonSerializerSettings SerializerSettings { get; set; } = DefaultSerializerSettings();
 
         /// <summary>
-        /// This option allows you to provide your own HttpMessageHandler. Useful with Android/iPhone.
+        /// Gets or sets the API version used for requests sent to Stripe's API. It is highly
+        /// recommended you do not change this value.
         /// </summary>
-        public static HttpMessageHandler HttpMessageHandler { get; set; }
-
-        public static TimeSpan? HttpTimeSpan { get; set; }
+        public static string StripeApiVersion { get; set; } = DefaultStripeApiVersion;
 
-        internal static string GetApiKey()
-        {
-#if NET45
-            if (string.IsNullOrEmpty(apiKey))
-            {
-                apiKey = System.Configuration.ConfigurationManager.AppSettings["StripeApiKey"];
-            }
-#endif
-
-            return apiKey;
-        }
+        /// <summary>Gets the version of the Stripe.net library.</summary>
+        public static string StripeNetVersion { get; }
 
-        public static void SetApiKey(string newApiKey)
+        /// <summary>
+        /// Gets or sets the timespan to wait before the request times out.
+        /// This property is deprecated and will be removed in a future version, please use the
+        /// <see cref="HttpTimeout"/> property instead.
+        /// </summary>
+        // TODO: remove this property in a future major version
+        [Obsolete("Use StripeConfiguration.HttpTimeout instead.")]
```

Comment:
We're already merging in a major version. Why not just do it?

Reply:
See above.